DFC.ServiceTaxonomy.Neo4j/Commands/DeleteNodesByTypeCommand.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using DFC.ServiceTaxonomy.Neo4j.Commands.Interfaces;
+using Neo4j.Driver;
+
+namespace DFC.ServiceTaxonomy.Neo4j.Commands
+{
+    public class DeleteNodesByTypeCommand : IDeleteNodesByTypeCommand
+    {
+        public HashSet<string> NodeLabels { get; set; } = new HashSet<string>();
+
+        public void CheckIsValid()
+        {
+            List<string> validationErrors = new List<string>();
+
+            if (!NodeLabels.Any())
+                validationErrors.Add($"Missing {nameof(NodeLabels)}.");
+
+            if (validationErrors.Any())
+                throw new InvalidOperationException(@$"{nameof(DeleteNodesByTypeCommand)} not valid:
+{string.Join(Environment.NewLine, validationErrors)}");
+        }
+
+        public Query Query
+        {
+            get
+            {
+                this.CheckIsValid();
+
+                return new Query(
+$@"MATCH (n:{string.Join(':', NodeLabels)})
+OPTIONAL MATCH (n)-[r]->()
+DELETE n, r");
+            }
+        }
+
+        public static implicit operator Query(DeleteNodesByTypeCommand c) => c.Query;
+
+        public void ValidateResults(List<IRecord> records, IResultSummary resultSummary)
+        {
+            //todo: What validation is possible here?
+        }
+
+        public List<string> ValidationErrors()
+        {
+            return new List<string>();
```

Comment:
This method needs to return the validation errors that the existing `CheckIsValid()` checks for, i.e. rename `CheckIsValid()` to `ValidationErrors()`, and return `validationErrors`, instead of throwing an exception.